bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/discovery/LutronDeviceDiscoveryService.java

```diff
@@ -288,14 +344,41 @@ private void processDevice(Device device, Stack<String> context) {
                     break;
 
                 case GRAFIK_EYE_QS:
-                    notifyDiscovery(THING_TYPE_GRAFIKEYEKEYPAD, device.getIntegrationId(), label);
+                    buttons = getComponentIdList(device.getComponents(), ComponentType.BUTTON);
+                    // remove button IDs >= 300 which the handler does not recognize
+                    List<Integer> buttonsCopy = new ArrayList<>(buttons);
+                    for (Integer c : buttonsCopy) {
+                        if (c >= 300) {
+                            buttons.remove(Integer.valueOf(c));
+                        }
+                    }
+                    kpConfig = new KeypadConfigGrafikEye();
+                    kpModel = kpConfig.determineModelFromComponentIds(buttons);
+                    if (kpModel == null) {
+                        logger.info("Unable to determine model of GrafikEye Keypad {} with button IDs: {}",
+                                device.getIntegrationId(), buttons);
+                        notifyDiscovery(THING_TYPE_GRAFIKEYEKEYPAD, device.getIntegrationId(), label);
+                    } else {
+                        logger.debug("Found GrafikEye keypad {} model: {}", device.getIntegrationId(), kpModel);
+                        notifyDiscovery(THING_TYPE_GRAFIKEYEKEYPAD, device.getIntegrationId(), label, "model", kpModel);
+                    }
                     break;
             }
         } else {
             logger.warn("Unrecognized device type {}", device.getType());
         }
     }
 
+    private List<Integer> getComponentIdList(List<Component> clist, ComponentType ctype) {
+        List<Integer> returnList = new LinkedList<>();
```

Comment:
I personally would prefer some JAva 8 kungfu for this ;-)

My IDE is updating so a quick one from the top of my mind

```java
return clist.stream().filter(c -> c.getComponentType() == ctype).collect(Collectors.toList());